pyensembl/genome.py

```diff
@@ -1041,3 +1042,25 @@ def protein_ids(self, contig=None, strand=None):
             distinct=True)
         # drop None values
         return [protein_id for protein_id in protein_ids if protein_id]
+
+    def __getstate__(self):
+        # Not the same as _fields(); these are useful for pickling/unpickling even if not necessary
+        # when checking for Genome equality.
+        field_list = [
+            "reference_name",
+            "annotation_name",
+            "annotation_version",
+            "_gtf_path_or_url",
+            "_transcript_fasta_path_or_url",
+            "_protein_fasta_path_or_url",
+            "decompress_on_download",
+            "copy_local_files_to_cache",
+            "require_ensembl_ids",
+            "cache_directory_path"]
```

Comment:
Ah, right: the annoying thing with `Genome` is that there are a ton of memoized fields that I don't want to save, and `memoized-property` doesn't provide an easy way to get at all of those. What do you think about rolling our own `memoized-property` that lets us easily access all of them and ignore them here?